src/then.js

```diff
@@ -1,38 +1,36 @@
+import Action from './Action'
+
 export default function then (f, r, p, promise) {
 	p._when(new Then(f, r, promise))
 	return promise
 }
 
-class Then {
+class Then extends Action {
 	constructor (f, r, promise) {
+		super(promise)
 		this.f = f
 		this.r = r
-		this.promise = promise
 	}
 
 	fulfilled (p) {
-		runThen(this.f, p, this.promise)
+		this.runThen(this.f, p)
 	}
 
 	rejected (p) {
-		return runThen(this.r, p, this.promise)
+		return this.runThen(this.r, p)
 	}
-}
 
-function runThen (f, p, promise) {
-	if (typeof f !== 'function') {
-		promise._become(p)
-		return false
+	runThen (f, p) {
+		if (typeof f !== 'function') {
+			this.promise._become(p)
+			return false
+		} else {
```

Comment:
I realize "no else after return" can be a controversial style, and I don't always do it.  I tend to prefer it in cases of checking something and returning early or taking a default action.  Since this code is so tiny, it's harder to justify that this is actually returning "early", but I do still prefer it without the else.